commonui/src/main/java/com/hyperwallet/android/ui/common/insight/HyperwalletInsight.java

```diff
@@ -216,21 +216,28 @@ public Handler getHandler() {
      * @param context      Context where the tracking is happening
      * @param pageName     an arbituary pageName the user is currently viewing as defined by the app
      * @param pageGroup    an arbituary pageGroup the user is currently viewing as defined by the app
+     * @param params    additional information that can be added to make the tracking event more useful
      * @param errorInfoMap contains information regarding the error
      */
     public void trackError(@NonNull final Context context, @NonNull final String pageName,
-            @NonNull final String pageGroup, @NonNull final Map<String, String> errorInfoMap) {
+            @NonNull final String pageGroup, @Nullable final Map<String, String> params,
+            @NonNull final Map<String, String> errorInfoMap) {
```

Comment:
We should keep only one map parameter (params and remove errorInfoMap) in this method. 
```suggestion
public void trackError(@NonNull final Context context, @NonNull final String pageName,
            @NonNull final String pageGroup, @Nullable final Map<String, String> params) {

        final ErrorInfo errorInfo = new ErrorInfo.ErrorInfoBuilder()
                .type(params.remove(InsightEventTag.InsightEventTagEventParams.ERROR_TYPE))
                .message(params.remove(InsightEventTag.InsightEventTagEventParams.ERROR_MESSAGE))
                .code(params.remove(InsightEventTag.InsightEventTagEventParams.ERROR_CODE))
                .field(params.remove(InsightEventTag.InsightEventTagEventParams.ERROR_FIELD_NAME))
                .description(params.remove(InsightEventTag.InsightEventTagEventParams.ERROR_DESCRIPTION))
                .build();
        
        if (Insight.getInsightTracker().isInitialized()) {
            Insight.getInsightTracker().trackError(context, pageName, pageGroup, params, errorInfo);

        } else {
            mExecutor.execute(new Runnable() {
                @Override
                public void run() {
                    Hyperwallet.getDefault().getConfiguration(new HyperwalletListener<Configuration>() {
                        @Override
                        public void onSuccess(@Nullable Configuration configuration) {
                            if (configuration != null) {
                                HyperwalletInsight.getInstance().initialize(context, configuration);
                                Insight.getInsightTracker().trackError(context, pageName, pageGroup, paramsForError,
                                        errorInfo);
                            }
                        }

                        @Override
                        public void onFailure(HyperwalletException exception) {
                            // do nothing
                        }

                        @Override
                        public Handler getHandler() {
                            return null;
                        }
                    });
                }
            });
        }
    }

```